### ISSUE TITLE
Access-Control-Allow-Origin

### ISSUE THREAD

**Issue by jesusto** [vitalidze/traccar-web#613]
I have a problem with Access-Control-Allow-Origin, install the traccar on port 80 centos linux, apache on port 8081, when I run the function, the console google gives me error

$.ajax({
    type: "POST",
    url: "http://149.56.3.246/traccar/rest/login",
    headers: {'Content-Type': 'application/x-www-form-urlencoded'},
    dataType: "json",
    data: "["+JSON.stringify(objJSON.user)+","+JSON.stringify(objJSON.password)+"]",
    success: function (res, status, xhr) { console.log("done");  console.log(res); console.log(status); console.log(xhr); 
             alert(xhr.getResponseHeader("Set-Cookie"));}
    });

**Comment by vitalidze** [comment]
Well, you need to either deploy your app under the same web server, or add Cors filter, something like this: https://github.com/orcoliver/traccar-web/commit/54db91a542366aa1de422ff4d04471097056e470

**Comment by jesusto** [comment]
a question, I have to modify the code and recompiling?

**Comment by vitalidze** [comment]
Yes, that's correct. You will need to add a java class, which will add necessary functionality. Easiest will be to add filter. Then you will need to build it by following official instructions: http://traccar.litvak.su/building.html (may require installation of Apache Maven)

**Comment by jesusto** [comment]
and which module I change?

**Comment by jesusto** [comment]
thank you, I just add, nobody calls?

**Comment by ghost** [comment]
Hi @vitalidze,

I really like what you did to traccar, thank you for your great contribution to humanity.

I have this small issue just like jesusto stated above, I did all the necessary things suggested on this thread, I copied src/main/java/org/traccar/web/server/model/CorsFilter.java to the same directory as my server copy, recompiled just as mentioned on this link http://traccar.litvak.su/building.html

I still can't get it to work :(

Here's my testing script.

http://54.254.236.138/traccar.php

Please help :)

**Comment by scriptologist** [comment]
@vitalidze

I also tested on the same server, same error :(
